PluginGenerator/PluginBuilder.cs

```diff
@@ -160,8 +210,101 @@ public void Build()
             BuildJar(tempWorkingDir);
         }
 
+        public string JarFilePath { get { return this.outputJarFilePath; } }
+
+        #endregion
+
         #region Private methods
 
+        private void Validate()
+        {
+            // TODO: validate other inputs
+            if (this.extensionClasses == null || !this.extensionClasses.Any())
+            {
+                throw new InvalidOperationException(UIResources.CoreBuilder_MustSpecifyAnExtensionClass);
+            }
+
+            if (string.IsNullOrWhiteSpace(this.FindPluginKey()))
+            {
+                throw new InvalidOperationException(UIResources.CoreBuilder_PluginKeyIsRequired);
+            }
+
+            if (string.IsNullOrWhiteSpace(this.FindPluginName()))
+            {
+                throw new InvalidOperationException(UIResources.CoreBuilder_PluginNameIsRequired);
+            }
+        }
+
+        private string FindPluginKey()
+        {
+            string pluginKey;
+            this.properties.TryGetValue(WellKnownPluginProperties.Key, out pluginKey);
+            return pluginKey;
+        }
+
+        private string FindPluginName()
+        {
+            string pluginName;
+            this.properties.TryGetValue(WellKnownPluginProperties.PluginName, out pluginName);
+            return pluginName;
+        }
+
+        private void AddCoreSources(string workingDirectory)
```

Comment:
Previously these .java files were added by the Rules builder. I've separated the Core and Rules resources into separate folders. The Rules builder now only adds rule-specific .java files and .jars.